### PR TITLE
chore: use py-arkworks's multi-exp method inside of `g1_lincomb` and `g2_lincomb`

### PR DESCRIPTION
**Pull request by kevaundray**
Motivation for this PR is to reduce CI times. Switch from double-and-add to the bucket method.

### CODE REVIEW

**Comment by kevaundray** [comment]
Copying @hwwhww previous profiling information:

https://github.com/ethereum/consensus-specs/pull/3713#issuecomment-2071421748

**Comment by kevaundray** [comment]
Tentatively, after this PR, the bottleneck switches to divide_polynomial_coefff -- there is a separate PR open to optimize that if we need to.

combined

**Comment by kevaundray** [comment]
Going to deduplicate the methods in bls.py

File: tests/core/pyspec/eth2spec/utils/bls.py [hunk 1]
```diff
@@ -225,6 +225,45 @@ def multiply(point, scalar):
     return py_ecc_mul(point, scalar)
 
 
+def multi_exp(points, integers):
+    """
+    Performs a multi-scalar multiplication between
+    `points` and `scalars`.
```
Comment (kevaundray) [inline tests/core/pyspec/eth2spec/utils/bls.py:231]:
```suggestion
    `points` and `integers`.
```

File: specs/deneb/polynomial-commitments.md [hunk 2]
```diff
@@ -146,6 +147,18 @@ def bit_reversal_permutation(sequence: Sequence[T]) -> Sequence[T]:
...
+#### `multi_exp`
+
+This function performs a multi-scalar multiplication between `points` and `integers`. `points` can either be in G1 or G2.
+
+```python
+def multi_exp(points: Sequence[TPoint],
+              integers: Sequence[uint64]) -> Sequence[TPoint]:
+    # pylint: disable=unused-argument
+    ...
+```
```
Comment (hwwhww) [inline specs/deneb/polynomial-commitments.md:160]: @kevaundray I added the abstract description like this (basically copied from your docstring). Feel free to update it.

**Review by hwwhww — APPROVED** [review]
lgtm 🚀🚀🚀

**Pull request merged**